client/webpack.config.js

```diff
@@ -187,6 +196,22 @@ module.exports = function exports(env, argv) {
           loader: 'ts-loader',
           exclude: /node_modules/,
         },
+        {
+          test: /axe\.js$/,
+          loader: path.resolve(__dirname, 'string-replaceall-loader.js'),
+          options: {
+            replace: {
+              "!('hasOwn' in Object)": 'false',
+              "!('values' in Object)": 'false',
+              "!('Promise' in window)": 'false',
+              "!('Uint32Array' in window)": 'false',
+              "!('some' in window.Uint32Array.prototype)": 'false',
+              "!('reduce' in window.Uint32Array.prototype)": 'false',
+              "typeof Object.assign !== 'function'": 'false',
+              'if (!Array.from)': 'if (false)',
+            },
```

Comment:
Unfortunately this doesn't do much – it will only remove the logic that does the actual polyfills, but not the imported polyfill code. This is because the imports happen inside the axe constructor, and the minifier can't detect that they are actually dead code.

I also tried adding the following (after looking into axe's package code) but it doesn't help:

```
              '__toModule(require_es6_promise())': 'null',
              '__toModule(require_typedarray())': 'null',
              '__toModule(require_weakmap_polyfill())': 'null',
              '__toModule(require_has_own3())': 'null',
              '__toModule(require_values3())': 'null',
              '__toModule(require_from4())': 'null',
```

Reply:
For more details:

The polyfills are imported at the top-level https://github.com/dequelabs/axe-core/blob/71cbd90fd3e5e0ded849d125b9a8b3ac064ff947/lib/core/imports/polyfills.js#L1-L6

and then in Axe's packaging process, webpack will put the imports inside a big function (that I assume to be a constructor of some kind) inside the main `axeFunction`:

![image](https://github.com/user-attachments/assets/1382ad8e-772f-4cca-8aa2-cc16a02b64ea)

the actual polyfilling happens later in the function:

![image](https://github.com/user-attachments/assets/7b3ad59d-c785-43e9-ac6c-a1fdd11aeddc)

Unfortunately it seems they're not that interested in making a polyfill-free version of Axe at the moment: https://github.com/dequelabs/axe-core/issues/1935